tests/cpptests/micro-benchmarks/benchmark_index_iterator.cpp

```diff
@@ -84,10 +84,14 @@ class BM_IndexIterator : public benchmark::Fixture {
     }
 
     void createIndex(IndexFlags flags) {
+        if (flags == (Index_DocIdsOnly | Index_Temporary)) {
+            // Special case reserved for `Index_DocIdsOnly` with raw doc IDs
+            RSGlobalConfig.invertedIndexRawDocidEncoding = true; // Enable raw doc ID encoding, until the benchmark's tearDown
```

Comment:
What happened with

```
RS_ASSERT_ALWAYS(encoder != InvertedIndex_GetEncoder(Index_DocIdsOnly)); // Ensure we are using the raw doc ID encoder
            encoder = InvertedIndex_GetEncoder(Index_DocIdsOnly);
            for (size_t i = 0; i < ids.size(); ++i) {
                RSIndexResult rec = {.docId = ids[i], .data = {.tag = RSResultData_Virtual}};
                InvertedIndex_WriteEntryGeneric(index, encoder, &rec);
            }
```

no longer required?

Reply:
That piece of code will be excercised by the `} else if (flags == Index_DocIdsOnly) {` case. Ie both the `else if` I removed and `else if` above it are now combined. This is possible because here we are setting the `RSGlobalConfig.invertedIndexRawDocidEncoding` early. This will cause the new combined case to correctly use the raw doc id encoding because of this

https://github.com/RediSearch/RediSearch/blob/07f7ed625f839328e68438aa75755a925053974c/src/inverted_index/inverted_index.c#L609-L614

Reply:
We can still keep the assertion here just to make sure this config is what is changing the behavior of `InvertedIndex_GetEncoder`